src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageItemListViewModel.cs

```diff
@@ -382,6 +470,271 @@ public bool IsPackageDeprecated
+                        {
+                            BitmapStatus = IconBitmapStatus.DefaultIconDueToNullStream;
+                        }
+                    }
+
+                    image = iconBitmapImage ?? Images.DefaultPackageIcon;
+                    // store this bitmapImage in the bitmap image cache, so that other occurences can reuse the BitmapImage
+                    string cacheKey = GenerateKeyFromIconUri(iconUrl);
+                    AddToCache(cacheKey, image);
+                    ErrorFloodGate.ReportAttempt();
+                }
+            }
+            return image;
+        }
+
+        private static void FinalizeBitmapImage(BitmapImage iconBitmapImage)
+        {
+            // Default cache policy: Per MSDN, satisfies a request for a resource either by using the cached copy of the resource or by sending a request
+            // for the resource to the server. The action taken is determined by the current cache policy and the age of the content in the cache.
+            // This is the cache level that should be used by most applications.
+            iconBitmapImage.UriCachePolicy = RequestCacheIfAvailable;
+
+            // Instead of scaling larger images and keeping larger image in memory, this makes it so we scale it down, and throw away the bigger image.
+            // Only need to set this on one dimension, to preserve aspect ratio
+            iconBitmapImage.DecodePixelWidth = DecodePixelWidth;
+
+            // Workaround for https://github.com/dotnet/wpf/issues/3503
+            iconBitmapImage.CreateOptions = BitmapCreateOptions.IgnoreColorProfile;
+
+            iconBitmapImage.CacheOption = BitmapCacheOption.OnLoad;
+
+            iconBitmapImage.EndInit();
+        }
+
+        /// <summary>
+        /// NuGet Embedded Icon Uri verification
+        /// </summary>
+        /// <param name="iconUrl">An URI to test</param>
+        /// <returns><c>true</c> if <c>iconUrl</c> is an URI to an embedded icon in a NuGet package</returns>
+        public static bool IsEmbeddedIconUri(Uri iconUrl)
+        {
+            return iconUrl != null
+                && iconUrl.IsAbsoluteUri
+                && iconUrl.IsFile
+                && !string.IsNullOrEmpty(iconUrl.Fragment)
+                && iconUrl.Fragment.Length > 1;
+        }
+
+        private static string GenerateKeyFromIconUri(Uri iconUrl)
+        {
+            return iconUrl == null ? string.Empty : iconUrl.ToString();
+        }
+
+        private static void AddToCache(string cacheKey, BitmapSource iconBitmapImage)
+        {
+            var policy = new CacheItemPolicy
+            {
+                SlidingExpiration = TimeSpan.FromMinutes(10),
+                RemovedCallback = CacheEntryRemoved
+            };
+            BitmapImageCache.Set(cacheKey, iconBitmapImage, policy);
+        }
+
+        private static void CacheEntryRemoved(CacheEntryRemovedArguments arguments)
```

Comment:
delete?

Reply:
it was wired up before, and was empty like this. 
but yes, likely removing the delegate set and the empty delegate will be a win.

Reply:
fixed in next iteration.